lizrd/grid/prepare_configs.py

```diff
@@ -0,0 +1,109 @@
+import hashlib
+from typing import List, Set, Tuple
+import yaml
+from lizrd.grid.infrastructure import prepare_default_infrastructure_params
+
+
+def get_yaml_md5(file_path):
+    with open(file_path, "rb") as f:
+        file_data = f.read()  # read file data into memory
+        hash_md5 = hashlib.md5(file_data).hexdigest()
+    return hash_md5
+
+
+def recursive_update(base_dict, update_dict):
+    for key, value in base_dict.items():
+        if isinstance(value, dict):
+            update_dict[key] = recursive_update(value, update_dict.get(key, {}))
+        elif key not in update_dict:
+            update_dict[key] = value
+    return update_dict
```

Comment:
I do believe it should be the other way around `base` is what should be modified using `update`, so update updates base. At least it's how I understand this naming. Please rename, for example `patch_dict` and `base_dict` or whatever you think makes sense here

Reply:
To pisane przez Szymona, na inną PR

Reply:
i tak robisz małą zmianę stylistyczną (zmiana miejsca tej funkcji), więc zmiana nazwy lokalnych parametrów (tylko w ramach tych 6 linijek) nie jest zmianą wymagającą osobnej PR moim zdaniem

Reply:
https://github.com/llm-random/llm-random/pull/374 dodałem osobną PR